docs/development/design/new-sphinx-guides.rst

```diff
@@ -0,0 +1,186 @@
+Proposed contents for new Sphinx guides
+=======================================
+
+This classification follows the `Diátaxis Framework <https://diataxis.fr/>`__.
+
+Tutorial
+--------
+
+   **Objective**: Write a tutorial, some examples, and reference
+   documentation for our Python code.
+
+Every step should lead to a “yay!” moment, which marks the end of the
+section. This keeps the learner motivated.
+
+Appendixes are optional, i.e. not required to follow the tutorial, but
+highly recommended.
+
+1. Getting started
+
+   1. Creating our code project
+
+      -  What is our library (present some minimal ``x.py`` file in a
+         dedicated directory, will be the basis of the tutorial)
+      -  Install Python
+      -  Check that our code works (yay!)
+
+   2. Using virtual environments for development
+
+      -  Create a virtual environment (and/or conda environment)
+      -  Activate our virtual environment (it will always be the first
+         step)
+      -  Check that our code also works (yay!)
+
+   3. Adding external dependencies
+
+      -  Add import from small third party library
+      -  Install it inside our virtual environment
+      -  Check that our code works (yay!) and it doesn’t outside the
+         virtual environment (as expected)
+
+   4. Appendix: Using version control
+
+      -  Install git (we will use it during the tutorial)
+      -  Write a barebones ``README.md`` (we will expand it later), a
+         ``.gitignore`` file (copied from gitignore.io), and a proper
+         ``LICENSE`` file
+
+         -  ``LICENSE.md``? ``LICENSE``? ``COPYING``? Whatever
+
+      -  Create the first commit for the project (yay!)
+
+2. First steps to document our project using Sphinx
+
+   1. Installing Sphinx
+
+      -  Activate our virtual environment
+      -  Install Sphinx inside the virtual environment
+      -  Check that ``sphinx-build --help`` works (yay!)
+
+   2. Creating the project layout
+
+      -  Run the sphinx-quickstart wizard: separate source and build
+         (``y``), project name, author name, and go
+      -  Check that the correct files are created (yay!)
```

Comment:
I feel like we probably need an explainer on how sphinx works and how to think about all the various files. Basically a longer version of this: https://sphinx-tutorial.readthedocs.io/start/#sphinx-philosophy

Similarly, do we want to introduce people to basic markdown? I'd be much more 👍  on explaining markdown to folks (and how Myst markdown is different) -- but we could also point folks to the Myst docs for that (https://myst-parser.readthedocs.io/en/latest/using/intro.html).  Perhaps a hoverxref of their page? :)

Should we start from a cookiecutter template instead of sphinx-quickstart? quickstart is kinda overwhelming.

Reply:
> I feel like we probably need an explainer on how sphinx works and how to think about all the various files. Basically a longer version of this: https://sphinx-tutorial.readthedocs.io/start/#sphinx-philosophy

Yes, in line with the Diátaxis Framework I considered this part of the "Explanations" (see below in the document).

> Similarly, do we want to introduce people to basic markdown? I'd be much more +1 on explaining markdown to folks (and how Myst markdown is different) -- but we could also point folks to the Myst docs for that (https://myst-parser.readthedocs.io/en/latest/using/intro.html). Perhaps a hoverxref of their page? :)

I think (and this applies to other comments as well), we should strike a balance between reinventing the wheel and adding too much content about things that are already explained elsewhere (because, well, they are already explained elsewhere) having too much indirection and pointing the learner to external resources too often, because this would reduce consistency and harm the learning experience, I believe. See for example what Astropy does with virtual environments:

https://docs.astropy.org/en/stable/development/workflow/virtual_pythons.html#virtual-envs

This is perhaps a bit too much for my taste, but I am not very happy with the other extreme, i.e. telling the user "before you continue, read this <here>" and not having any control over that material.

> Should we start from a cookiecutter template instead of sphinx-quickstart? quickstart is kinda overwhelming.

I also considered this, and I was inclined to use `sphinx-quickstart` instead. Years ago it asked you about every other plugin and it was quite intimidating as you say, but these days it only asks about five questions.

Reply:
- Added a section at the beginning give more background on Sphinx, Markdown, and MyST
- Changed proposal to use a cookiecutter (we would need to create it)
- Removed mentions to reST in favor of Markdown